Shopify/cli
# Move some instrumentation into `app` from `cli-kit`

### WHY are these changes introduced?

Ensures that app specific metadata comes from `@shopify/app`



### WHAT is this pull request doing?



Our first usage of the new CLI metadata foundations -- uses a hook to gather data from plugins.

### How to test your changes?



`DEBUG="*" yarn shopify app dev --path=xxx` 
- you'll see the analytics payload includes project type, partner ID and api key
- you'll see hooks were loaded and run:
<img width="598" alt="image" src="https://user-images.githubusercontent.com/234027/182636078-79427afb-b1e3-478f-a056-0cb7fecba01b.png">
